protocol/flows/src/v5/ibd/flow.rs

```diff
@@ -146,14 +146,30 @@ impl IbdFlow {
                 return Ok(IbdType::Sync(highest_known_syncer_chain_hash));
             }
 
-            // TODO: in this case we know a syncer chain block, but it violates our current finality. In some cases
-            // this info should possibly be used to reject the IBD despite having more blue work etc.
+            // If the pruning point is not in the chain of `highest_known_syncer_chain_hash`, it
+            // means it's in its antichain (because if `highest_known_syncer_chain_hash` was in
+            // the pruning point's past the pruning point itself would be
+            // `highest_known_syncer_chain_hash`). So it means there's a finality conflict.
+            // TODO: Find a better way to handle finality conflicts.
```

Comment:
1. `TODO: Find a better way to handle finality conflicts.` appears 3 times in this file now. I think once in the correct location is enough 
2. while we are at it, there are 2 more simple `TODO` checks in this file which we should cover:
  `TODO: make sure that the proof pruning point is not genesis`
  `TODO: verify the proof pruning point is different than current consensus pruning point`